packages/liveblocks-client/src/__tests__/AuthToken.test.ts

```diff
@@ -36,24 +36,20 @@ describe("isTokenExpired", () => {
 
 describe("parseRoomAuthToken", () => {
   const roomToken =
-    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2MTY3MjM2NjcsImV4cCI6MTYxNjcyNzI2Nywicm9vbUlkIjoiazV3bWgwRjlVTGxyek1nWnRTMlpfIiwiYXBwSWQiOiI2MDVhNGZkMzFhMzZkNWVhN2EyZTA5MTQiLCJhY3RvciI6MCwic2NvcGVzIjpbIndlYnNvY2tldDpwcmVzZW5jZSIsIndlYnNvY2tldDpzdG9yYWdlIiwicm9vbTpyZWFkIiwicm9vbTp3cml0ZSJdfQ.IQFyw54-b4F6P0MTSzmBVwdZi2pwPaxZwzgkE2l0Mi4";
+    "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NjQ1NjY0MTAsImV4cCI6MTY2NDU3MDAxMCwicm9vbUlkIjoiS1hhNlVjbHZyYWVHWk5kWFZ6NjdaIiwiYXBwSWQiOiI2MDVhNGZkMzFhMzZkNWVhN2EyZTA4ZjEiLCJhY3RvciI6ODcsInNjb3BlcyI6WyJyb29tOndyaXRlIl19.uS0VcdeAPdMfJ2rseRRUnL_X3I-h6ljPKEiu1xfKRG0Qrth0zdqo2ngn7NZ8_fLcQBaIvaZ4q5vXg_Nex81Ae9sjmmLhjxHcE-iA-BC82NROVSnyGdVHJRMNqs6h57pCdiXwCwpcLjqi_EOIS8gmMB8dcRX748Wpa4C2T0e94An8_vP6eD66JKndxjFvVPrB_LSOOlQZoxW9USPS7ZUTAECeGQscrXnss_-1TJEaGf0RxVkNQsDfUKu4TjWYa3iBvBPip--Ev1bBETh0IHrGNsWVUd-691cCRAemiC_ADBaOg5IEszqoEw96Xe9BtQeWrjAgMKKrPS72cwkikVmiJQ";
+
   const apiToken =
     "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2MTY3MjM2NjcsImV4cCI6MTYxNjcyNzI2NywiYXBwSWQiOiI2MDVhNGZkMzFhMzZkNWVhN2EyZTA5MTQiLCJzY29wZXMiOlsicm9vbTpyZWFkIiwicm9vbTp3cml0ZSJdfQ.7Wplt6YV_YbpPAcAFyC8pX8tk5BGNy53GdoH1_u8sjo";
 
   test("should parse a valid token", () => {
     const parsedToken = parseRoomAuthToken(roomToken);
     expect(parsedToken).toEqual({
-      iat: 1616723667,
-      exp: 1616727267,
-      roomId: "k5wmh0F9ULlrzMgZtS2Z_",
-      appId: "605a4fd31a36d5ea7a2e0914",
-      actor: 0,
-      scopes: [
-        "websocket:presence",
-        "websocket:storage",
-        "room:read",
-        "room:write",
-      ],
+      actor: 87,
+      appId: "605a4fd31a36d5ea7a2e08f1",
+      exp: 1664570010,
+      iat: 1664566410,
+      roomId: "KXa6UclvraeGZNdXVz67Z",
+      scopes: ["room:write"],
```

Comment:
I guess what I'm lacking is an understanding of which scopes are implicit. If you have `["room:write"]` scope, what permissions do you have for presence? Are `"room:read"` and `"room:presence:read"` implied? Or does `["room:write"]` explicitly disallow read permissions everywhere, because those scopes aren't in there?

Reply:
Good point, if you're asking the question we need clarification somewhere.


- `room:write` = `room:storage:write` + `room:presence:write`
    and the same logic would apply for `room:read`

- `room:write` = `room:write` + `room:read`
    this is an implicit simplification so that users are not required to add `read` whenever something is writable since writing is a stronger permission than reading
    
The simplifications are made so that less permissions are required to be set by our users

Let me know if that makes sense 🙏

Reply:
Aha! That's good to know.

So, are the following conclusions correct?

- `["room:storage:read"]` expands to `["room:storage:read"]`
- `["room:storage:write"]` expands to `["room:storage:read", "room:storage:write"]`
- `["room:presence:read"]` expands to `["room:presence:read"]`
- `["room:presence:write"]` expands to `["room:presence:read", "room:presence:write"]`
- `["room:write"]` expands to `["room:storage:read", "room:storage:write", "room:presence:read", "room:presence:write"]`
- `["room:read"]` expands to `["room:storage:read", "room:presence:read"]` (but no write permissions)
- `["room:write", "room:storage:read"]` expands to `["room:storage:read", "room:storage:write", "room:presence:read", "room:presence:write"]` (the explicit read perms are unnecessary?)

If so, what does "nothing" expand to?

- `[]` expands to `[]`? So no access whatsoever?

Reply:
Exact! 

Except that we'll allow at the API level only two options for now: `room:write` or `room:read` + `room:presence:write`

`[]` will not be possible as basically if the user has no access, then it gets a 403 when trying to obtain a token.

But yes, `[]` would expand to `[]` if we were to emit a token (which we do not)

Reply:
Amazing! That is great, and makes so much sense! Thanks for clarifying.